### PR TITLE
fix(texlab): use exec_cmd method

### PR DESCRIPTION
**Pull request by DimitrisDimitropoulos**
Problem: vim.lsp.buf.execute_command has been deprecated

For more info see #3426.

### CODE REVIEW

**Comment by glepnir** [comment]
can simply like this ?

```diff
diff --git a/lua/lspconfig/configs/texlab.lua b/lua/lspconfig/configs/texlab.lua
index c1a5757..b1a4aaf 100644
--- a/lua/lspconfig/configs/texlab.lua
+++ b/lua/lspconfig/configs/texlab.lua
@@ -52,9 +52,16 @@ end

 local function buf_cancel_build()
   local bufnr = vim.api.nvim_get_current_buf()
-  if not util.get_active_client_by_name(bufnr, 'texlab') then
+  local client = util.get_active_client_by_name(bufnr, 'texlab')
+  if not client then
     return vim.notify('Texlab client not found', vim.log.levels.ERROR)
   end
+  if vim.fn.has('nvim-0.11') == 1 then
+    return client:exec_cmd({
+      title = 'cancel',
+      command = 'texlab.cancelBuild',
+    }, { bufnr = bufnr })
+  end
   vim.lsp.buf.execute_command { command = 'texlab.cancelBuild' }
   vim.notify('Build cancelled', vim.log.levels.INFO)
```

**Comment by DimitrisDimitropoulos** [comment]
adapted it to suggested style

File: lua/lspconfig/configs/texlab.lua [hunk 1]
```diff
@@ -1,4 +1,5 @@
 local util = require 'lspconfig.util'
+local nvim_eleven = vim.fn.has 'nvim-0.11' == 1
```
Comment (glepnir) [inline lua/lspconfig/configs/texlab.lua:2]:
```suggestion
```

File: lua/lspconfig/configs/texlab.lua [hunk 2]
```diff
@@ -52,9 +53,15 @@ end
 
 local function buf_cancel_build()
   local bufnr = vim.api.nvim_get_current_buf()
-  if not util.get_active_client_by_name(bufnr, 'texlab') then
+  local client = util.get_active_client_by_name(bufnr, 'texlab')
+  if not client then
     return vim.notify('Texlab client not found', vim.log.levels.ERROR)
   end
+  if nvim_eleven == 1 then
```
Comment (glepnir) [inline lua/lspconfig/configs/texlab.lua:60]:
```suggestion
  if vim.fn.has 'nvim-0.11' == 1 then
```

File: lua/lspconfig/configs/texlab.lua [hunk 3]
```diff
@@ -52,9 +52,15 @@ end
...
 local function buf_cancel_build()
   local bufnr = vim.api.nvim_get_current_buf()
-  if not util.get_active_client_by_name(bufnr, 'texlab') then
+  local client = util.get_active_client_by_name(bufnr, 'texlab')
+  if not client then
     return vim.notify('Texlab client not found', vim.log.levels.ERROR)
   end
+  if vim.fn.has 'nvim-0.11' == 1 then
+    return client:exec_cmd({
+      command = 'texlab.cancelBuild',
```
Comment (glepnir) [inline lua/lspconfig/configs/texlab.lua:61]:
```suggestion
    return client:exec_cmd({
      title = 'cancel',
      command = 'texlab.cancelBuild',
```

Comment (DimitrisDimitropoulos) [reply lua/lspconfig/configs/texlab.lua]: Does `exec_cmd` support a `title` argument? Based upon documentation this is not clear

Comment (glepnir) [reply lua/lspconfig/configs/texlab.lua]: it's a require field.

QQ_1731233778465

**Comment by DimitrisDimitropoulos** [comment]
Is there something blocking this PR from getting merged? Should all changes per server be included in this PR, or would it be better to split them up into multiple PRs?

**Pull request merged**